presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java

```diff
@@ -0,0 +1,183 @@
+package com.facebook.presto.operator.scalar;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.facebook.presto.Session;
+import com.facebook.presto.SessionTestUtils;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+import com.google.common.io.BaseEncoding;
+import io.airlift.slice.Slice;
+import io.airlift.stats.cardinality.HyperLogLog;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+public class TestHyperLogLogFunctions
+        extends AbstractTestFunctions
+{
+    private TestHyperLogLogFunctions() {}
+
+    private static final int NUMBER_OF_BUCKETS = 32768;
+
+    @Test
+    public void testApproxNullArray()
+    {
+        String projection = "cardinality(merge_hll(NULL))";
+
+        Session session = SessionTestUtils.TEST_SESSION;
+        FeaturesConfig featuresConfig = new FeaturesConfig();
+        LocalQueryRunner runner = new LocalQueryRunner(session, featuresConfig);
+
+        MaterializedResult result = runner.execute("SELECT " + projection);
+
+        assertNull(result.getOnlyValue());
+    }
+
+    @Test
+    public void testApproxMultipleNullColumns()
+    {
+        String projection = "cardinality(merge_hll(ARRAY[null, null, null]))";
+
+        Session session = SessionTestUtils.TEST_SESSION;
+        FeaturesConfig featuresConfig = new FeaturesConfig();
+        LocalQueryRunner runner = new LocalQueryRunner(session, featuresConfig);
+
+        MaterializedResult result = runner.execute("SELECT " + projection);
+
+        assertNull(result.getOnlyValue());
+    }
+
+    @Test
+    public void testApproxNoColumns()
+    {
+        long blockSize = 0;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getProjectionString(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        assertNull(getCardinality(projection));
+    }
+
+    @Test
+    public void testApproxSingleColumn()
+    {
+        long blockSize = 1;
+        long uniqueElts = 10000 * blockSize;
+
+        String projection = getProjectionString(getUniqueElements((int) blockSize, uniqueElts));
+
+        double error = uniqueElts * 0.05;
+        assertEquals(getCardinality(projection), uniqueElts, error);
```

Comment:
We can create a `assertFunction` which will internally do the equals-with-delta.  And we could consider putting this in `AbstractTestFunctions`.  I.e. the signature may look like:

`void assertFunction(String projection, Type expectedType, Number expectedNumber, double delta)`